### PR TITLE
Improve Dockerfile and update docker-compose.yml

### PR DESCRIPTION
**Pull request by crohr**
Cleaning pass on the Dockerfile, removing any unneeded files, and using multi-stage build to compile pgloader-ccl. Saved about 300MB on the resulting image.

Also added an upgraded docker-compose.yml file, using latest version of the spec (v3.7), with one service per container.

This means the docker image works in both the all-in-one approach, and as well in the new single-process per container approach.

### CODE REVIEW

**Comment by ulferts** [comment]
@crohr there was a problem with the localization files during the time you branched. Mergin/Rebasing to the current dev should fix the problems.

File: docker/proxy.conf.erb [hunk 1]
```diff
@@ -5,7 +5,7 @@
   ProxyRequests off
 
   <Location />
-    ProxyPass http://127.0.0.1:8080/ retry=0
-    ProxyPassReverse http://127.0.0.1:8080/
+    ProxyPass http://<%= ENV.fetch("APP_HOST", "127.0.0.1") %>:8080/ retry=0
+    ProxyPassReverse http://<%= ENV.fetch("APP_HOST", "127.0.0.1") %>:8080/
```
Comment (olivierdalang) [inline docker/proxy.conf.erb]: I don't see where APP_HOST comes from ? Also not sure the change is related, as I don't see how apache is used in your setup ?

File: docker-compose.yml [hunk 2]
```diff
@@ -1,30 +1,76 @@
...
+
+  cron:
+    <<: *app
+    command: "./docker/cron"
+    networks:
+      - backend
+
+  seeder:
+    <<: *app
+    command: "./docker/seeder"
```
Comment (olivierdalang) [inline docker-compose.yml:87]: I don't see the seeder script ?

**Review by olivierdalang — COMMENTED** [review]
Hi ! I think it's already a good improvement.

Here are a few comments, some of which are implemented in [this branch](https://github.com/olivierdalang/openproject/tree/docker-compose) with similar goals (it's a WIP, I'm not done with all services yet, and of course I'd rather concentrate efforts with yours).

#### reverse proxy
Am I right understanding there's no reverse proxy that setup (I don't see a container for apache despite some changes to the apache config) ? Maybe we could use traefik as a reverse proxy, which is nice as completely configurable in docker-compose, and provides out of the box https encryption with let's encrypt.

#### configuration
How would one configure the stack (e.g. port/host name or postgres db/password) ? I'd suggest using the .env with docker-compose variable substitution for all the main user settings, so that user-dependant settings aren't spread over the docker-compose file.

#### postfix 
Wouldn't this also deserve it's own service ?

#### general setup
As said in the issue, I'd be in favour of removing the supervisord setup from the docker image altogether, as to optimize image size and make the setup more understandable/maintainable. If we publish the image under a new tag, that won't break existing workflows, and I think it's safer to do this at the same time than before officially publishing a docker-compose setup rather than afterwards, as it already supposes changes in the deploy method. We could keep two Dockerfiles in parralel for some deprecation time if needed.

**Comment by olivierdalang** [comment]
(forgot to mention I didn't test the branch yet, just looked at the changes)

**Comment by oliverguenther** [comment]
I merged in dev to resolve the conflicts. your changes appear to be fully contained in `docs/` but please double check

**Review by machisuji — APPROVED** [review]
Looks good, thanks @crohr!

**Comment by machisuji** [comment]
Thanks for resolving the conflict @oliverguenther ! <3

**Comment by machisuji** [comment]
I've tested both the docker-compose setup and the standalone container, and both with or without RAILS_RELATIVE_URL_ROOT. It works like a charm.

@olivierdalang made some valid comments which we may address separately. For now I prefer to keep the image working in both scenarios under the same tag. But yes, perhaps we could add a slimmer image under a separate tag too.

Either way I'm merging this now.

**Pull request merged**